### PR TITLE
Support initialize/setup

### PR DESCRIPTION
**Pull request by jeffreyssmith2nd**
Note: Won't build until [Demux PR 116](https://github.com/EOSIO/demux-js/pull/116) is merged and the dependent version here is updated

### CODE REVIEW

File: src/NodeosActionReader.ts [hunk 1]
```diff
@@ -78,4 +78,19 @@ export class NodeosActionReader extends AbstractActionReader {
...
+      return
+    }
+
+    try {
+      await request.get({
+        url: `${this.nodeosEndpoint}/v1/chain/get_info`,
+        json: true,
+      })
+    } catch (err) {
+      throw new NotInitializedError('Cannot reach supplied nodeos endpoint.')
```
Comment (flux627) [inline src/NodeosActionReader.ts]: Maybe we shouldn't swallow the underlying error here for better debugging.

File: src/testHelpers/mongoMock.ts [hunk 2]
```diff
@@ -1,134 +1,72 @@
...
-    confirm_count: [],
-    confirmations: [],
-  },
-  block_num: 21,
-  in_current_chain: true,
-  validated: true,
+import { actionTraces } from './actionTraces'
+import { blockStates } from './blockStates'
+
+const deepCloneBlockStates = (bStates: any) => {
```
Comment (flux627) [inline src/testHelpers/mongoMock.ts]: Does `deepCloneBlockState` not work on arrays as well?

Comment (jeffreyssmith2nd) [reply src/testHelpers/mongoMock.ts]: Good call

File: src/NodeosActionReader.ts (hunk 1 above)
Comment (jeffreyssmith2nd) [reply src/NodeosActionReader.ts]: I thought about that, but then the client code can't do initialization specific logic solely based on `NotInitializedErrors`. That may be okay, in which case I can throw the underlying error, but that's why I did it this way

Comment (flux627) [reply src/NodeosActionReader.ts]: So, if they wrap `initialize` in a try/catch, then any error that they catch can be assumed to be an initialization error, right?

Comment (jeffreyssmith2nd) [reply src/NodeosActionReader.ts]: The initial thought I had was you could do specific `typeof NotInitializedError` stuff if you want, but yeah there's no reason it couldn't throw other errors and those would still be caught by a generic `try/catch`.

**Pull request merged**